lib/index.js

```diff
@@ -168,12 +171,19 @@ class PluginPassportOAuth {
     return this.getProviderRepository(profile.provider)
       .get(profile._json[this.config.strategies[profile.provider].identifierAttribute])
       .then(userObject => {
-        let req = null;
-
         if (userObject !== null) {
           return Promise.resolve({kuid: userObject.kuid, message: null});
         }
 
+        return Promise.reject({message: `Could not login with strategy "${profile.provider}"`});
+      })
+      .catch(err => {
+        if (!(err instanceof NotFoundError)) {
+          return Promise.reject(err);
+        }
+
+        let req = null;
```

Comment:
Should be made const, and declared in the innermost scope possible. Here, it would be line 194.